crates/bevy_render/macros/src/as_bind_group.rs

```diff
@@ -598,43 +598,39 @@ const VISIBILITY_COMPUTE: Symbol = Symbol("compute");
 const VISIBILITY_ALL: Symbol = Symbol("all");
 const VISIBILITY_NONE: Symbol = Symbol("none");
 
-fn get_visibility_flag_value(
-    nested_metas: &Punctuated<NestedMeta, Token![,]>,
-) -> Result<ShaderStageVisibility> {
+fn get_visibility_flag_value(meta: Meta) -> Result<ShaderStageVisibility> {
     let mut visibility = VisibilityFlags::vertex_fragment();
 
-    for meta in nested_metas {
-        use syn::{Meta::Path, NestedMeta::Meta};
-        match meta {
-            // Parse `visibility(all)]`.
-            Meta(Path(path)) if path == VISIBILITY_ALL => {
-                return Ok(ShaderStageVisibility::All)
-            }
-            // Parse `visibility(none)]`.
-            Meta(Path(path)) if path == VISIBILITY_NONE => {
-                return Ok(ShaderStageVisibility::None)
-            }
-            // Parse `visibility(vertex, ...)]`.
-            Meta(Path(path)) if path == VISIBILITY_VERTEX => {
-                visibility.vertex = true;
-            }
-            // Parse `visibility(fragment, ...)]`.
-            Meta(Path(path)) if path == VISIBILITY_FRAGMENT => {
-                visibility.fragment = true;
-            }
-            // Parse `visibility(compute, ...)]`.
-            Meta(Path(path)) if path == VISIBILITY_COMPUTE => {
-                visibility.compute = true;
-            }
-            Meta(Path(path)) => return Err(Error::new_spanned(
-                path,
-                "Not a valid visibility flag. Must be `all`, `none`, or a list-combination of `vertex`, `fragment` and/or `compute`."
-            )),
-            _ => return Err(Error::new_spanned(
-                meta,
-                "Invalid visibility format: `visibility(...)`.",
-            )),
+    use syn::Meta::Path;
+    match meta {
+        // Parse `visibility(all)]`.
+        Path(path) if path == VISIBILITY_ALL => {
+            return Ok(ShaderStageVisibility::All)
+        }
+        // Parse `visibility(none)]`.
```

Comment:
```suggestion
        // Parse `#[visibility(all)]`.
        Path(path) if path == VISIBILITY_ALL => {
            return Ok(ShaderStageVisibility::All)
        }
        // Parse `#[visibility(none)]`.
```

I know the formatting was already messed up on these comments but we might as well fix it here.